pkg/storage/filepath/jsonfile_rest.go

```diff
@@ -234,6 +237,22 @@ func (f *filepathREST) Update(
 			return nil, false, err
 		}
 	}
+
+	// handle 2-phase deletes -> for entities with finalizers, DeletionTimestamp is set and reconcilers execute +
+	// remove them (triggering more updates); once drained, it can be deleted from the final update operation
+	// loosely based off https://github.com/kubernetes/apiserver/blob/947ebe755ed8aed2e0f0f5d6420caad07fc04cc2/pkg/registry/generic/registry/store.go#L624
+	objMeta := updatedObj.(resource.Object).GetObjectMeta()
```

Comment:
i think you can use meta.Accessor() here? https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#Accessor

it's pretty handy